contest/contestArchive.py

```diff
@@ -0,0 +1,63 @@
+'''
+    The MIT License (MIT)
+    Copyright (c) 2014 NTHUOJ team
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    '''
+from datetime import datetime
+from django.db.models import Q
+from contest.models import Contest 
+from contest.models import Contestant
+
+def get_contests(user):
+    if user.is_authenticated():
+        if user.has_admin_auth():
+            #admin show all
+            contests_info = Contest.objects.order_by('-start_time')
+        elif user.has_subjudge_auth():
+            contests_info = get_owned_or_started_contests(user)
+        else:
+            contests_info = get_started_contests()
+    else:
+        #user not logged in
+        contests_info = get_started_contests()
+
+    contests = []
+    for contest in contests_info:
+        new_contest = add_contestants(contest)
+        contests.append(new_contest)
+    return contests
+
+def get_owned_or_started_contests(user):
+    owned_contests = get_owned_contests(user)
+    started_contests = get_started_contests()
+    return owned_contests | started_contests
```

Comment:
The object is QuerySet
which does not have method extend
I found this way on stack overflow
http://stackoverflow.com/questions/431628/how-to-combine-2-or-more-querysets-in-a-django-view